src/nodes.cpp

```diff
@@ -23,6 +25,91 @@ NodePopulation::NodePopulation(const std::string& h5FilePath,
                                const std::string& name)
     : Population(h5FilePath, csvFilePath, name, ELEMENT) {}
 
+
+namespace {
+template <typename T>
+Selection _getMatchingSelection(const std::vector<T>& values, const T value) {
+    Selection::Values idx;
+    Selection::Value id = 0;
+
+    for (const auto& v : values) {
+        if (v == value) {
+            idx.push_back(id);
+        }
+        ++id;
+    }
+    return Selection::fromValues(idx);
+}
+
+template <typename T>
+Selection _matchAttributeValues(const NodePopulation& population,
+                                const std::string& name,
+                                T value) {
+    return _getMatchingSelection(population.getAttribute<T>(name, population.selectAll()), value);
+}
+
+
+Selection _matchAttributeValues(const NodePopulation& population,
+                                const std::string& name,
+                                std::string value) {
+    // enum attribute, need to look up in the @library
+    if (population.enumerationNames().count(name) > 0) {
+        const auto enum_values = population.enumerationValues(name);
+        const auto pos = std::find(enum_values.begin(), enum_values.end(), value);
+        if (pos == enum_values.end()) {
+            return Selection({});
+        }
+        return _getMatchingSelection<size_t>(
+            population.getEnumeration<size_t>(name, population.selectAll()),
+            pos - enum_values.begin());
+    }
+
+    // normal attribute
+    return _matchAttributeValues<std::string>(population, name, value);
+}
+
+}  // anonymous namespace
+
+template <typename T>
+Selection NodePopulation::matchAttributeValues(const std::string& name, const T value) const {
```

Comment:
I'm not really happy w/ this method of dispatch.  Any suggestions are very welcomed.

Basically, I need `typename T` to be comparible to whatever the `HighFive::AtomicType` is - which is dynamic from whatever is loaded.  Currently, I think this might give surprising results if the value is a `float` of 1.5, but the attribute column type is int -> a rounding could happen and you'd get a Selection that is surprising.  Or maybe I'm seeing ghosts.

Reply:
I think that's a good concern. Should at least be tested. I guess that `_matchAttributeValues` should be called with the HighFive-determined datatype?

It would be a big ugly dispatch table @tristan0x may have a more elegant solution.

Reply:
Actually even `float` to `float` comparison seems dangerous. Wouldn't it be better to not use `float` with `matchAttribteValues` at all but define a selector based on `operator<` for `float` ?

Reply:
So, this is in support of node_sets, which only define equality comparisons.  I considered only doing integer comparisons, but I can (barely) see where floating point comparisons could be useful, so I added those too.

Reply:
It feels unreliable to me, or at least use a pre-set tolerance so that people know what to expect.

Reply:
I don't think having a tolerance is the best plan, especially since this will be called based off the node_set json definition - ie: not by someone reading the docstring.

Reply:
I think not supporting float conversions then. I thought also that H5 converts types under the hood?

If something is written to disk as `float` and implicitly converted to `double`, will the representation be equal?

Reply:
> I think not supporting float conversions then. I thought also that H5 converts types under the hood?

You mean double -> float? Or just not having float compare at all?  The latter was my original 'solution' to the whole problem, as I'm not convinced of the utility of it, however the spec says that `node_sets` can have `json number` -> `H5T_IEEE_*LE, H5T_STD_*LE`. So I feel boxed in.

> If something is written to disk as float and implicitly converted to double, will the representation be equal?

You mean if a float is widened to a double? Yes, they are the same (as per conv.fpprom in c++ standardeze).